src/MessagingAvatar/MessagingAvatar.tsx

```diff
@@ -11,7 +11,7 @@ const cssClasses = {
 type Props = {
   className?: string;
   text: string;
-  color?: { color: string } | { seed: string };
+  color?: { color: string, seed?: undefined } | { seed: string, color?: undefined };
```

Comment:
I didn't realize this at the time I suggested it, but it turns out the `A | B` approach isn't sufficient for making mutually exclusive props. Previously, something like `{ color: "fff", seed: "abc" }` would have still been allowed even though it's really supposed to be one or the other. The fix is to explicitly mark the unused props as "undefined" or "never". 

This seems reasonable right now that there's only two props, but if we add more I think it's a fair question whether it's worth the effort to maintain the mutual exclusiveness. Either way, I think sticking with it for now is fine and we can always make it more permissive later without it being a breaking change.